rasa/core/actions/forms.py

```diff
@@ -101,6 +104,31 @@ def get_mappings_for_slot(
 
         return requested_slot_mappings
 
+    def _create_unique_entity_mappings(self, domain: Domain) -> Set[Text]:
+        unique_entity_slot_mappings = set()
```

Comment:
Why should it happen that we have a duplicated slot mapping? Can you give an example?

Reply:
if we have the following slot_mappings:
```    departure_city:
      - type: from_entity
        entity: city
        role: from
      - type: from_entity
        entity: city
    arrival_city:
      - type: from_entity
        entity: city
        role: to
      - type: from_entity
        entity: city
    arrival_date:
      - type: from_entity
        entity: date
```
then 
```
      - type: from_entity
        entity: city
        role: from
      - type: from_entity
        entity: city
        role: to
      - type: from_entity
        entity: date
```
are unique, because they don't fill different slots. However
```
      - type: from_entity
        entity: city
```
is not unique it fills different slots depending on which one is requested

Reply:
Ah ok, that makes sense, was missing that we don't care about the actual slot to fill in this check.

Should we add a comment like:
``` 
A slot mapping of type "from_entity" is unique if there is no other slot mapping with the same conditions. For example:
Given the slot mappings 
  destination_city:
    - type: from_entity
      entity: city
departure_city:
    - type: from_entity
      entity: city
the slot mapping
    - type: from_entity
      entity: city
is not unique as it is used for two different slots.
```

Reply:
added explanation in a docstring